rustls/src/crypto/mod.rs

```diff
@@ -43,6 +46,14 @@ pub trait CryptoProvider: Send + Sync + Debug + 'static {
 
     /// Return a safe set of supported key exchange groups to be used as the defaults.
     fn default_kx_groups(&self) -> &'static [&'static dyn SupportedKxGroup];
+
+    /// Decode and validate a private signing key from `key_der`.
+    ///
+    /// Return an error if the key type encoding is not supported, or if the key fails validation.
+    fn load_private_signing_key(
```

Comment:
Is the `_signing` part important here? Feels like we could leave that to documentation? (Alternatively, should we rename `pki_types::PrivateKeyDer<'_>` to `PrivateSigningKeyDer<'_>`?

Reply:
i'll drop `_signing` but keep it in the docs. i don't foresee a need to handle non-signing private keys; that would imply we're doing some non-ephemeral DH or RSA encryption things, both of which I think are far outside what we should be concerned about.